04.ls/ls.rb

```diff
@@ -1,18 +1,42 @@
 #!/usr/bin/env ruby
 
-dir = Dir.glob '*'
-FILES = dir.count
-COLUMN = 3
-ROW = FILES.ceildiv(COLUMN)
-
-MAX_ROW = dir.max_by(&:length).length + 2
-
-index = 0
-1.upto(ROW) do |row| # 行ごとに繰り返し
-  COLUMN.times do # 列ごとに繰り返し
-    print dir[index].to_s.ljust(MAX_ROW, ' ') unless dir[index].nil?
-    index += ROW # 行数の倍数 == 配列から取得するindex変数の値
+require 'optparse'
+require 'etc'
+
+option = nil
+ARGV.options do |opt|
+  opt.on('-l') { option = '-l' }
+  opt.parse!
+end
+
+def l_option
+  Dir.glob('*').each do |name|
+    info = File.stat(name)
+    mode = info.mode.to_s(8)[-4..-1]
+    mode = mode.gsub('7', 'rwx').gsub('6', 'rw-').gsub('5', 'r-x').gsub('4', 'r--').gsub('3', '-wx').gsub('2', '-w-').gsub('1', '--x').tr('0', '-')
```

Comment:
こちら、1行が長いのでrubocopから指摘が出ないでしょうか？
7〜0を全てベタ書きしている状態ですが、繰り返し処理を用いる等してさらに効率的な書き方がないか、検討いただけますでしょうか？

Reply:
Rubocopで指摘される文字数を調べてみると160文字でしたので、今回は140文字程度だったので指摘はされていないようでした。
繰り返し処理を用いて次のように修正をしてみました。
```suggestion
def chmod(mode)
  mode.chars.map do |num|
    %w(- --x -w- -wx r-- r-x rw- rwx)[num.to_i]
  end.join
end

def l_option
  Dir.glob('*').each do |name|
    info = File.stat(name)
    mode = info.mode.to_s(8)[-4..-1]
    chmod = chmod(mode)
```